src/client.ts

```diff
@@ -1,499 +1,424 @@
+            reject(err)
+            return
+          }
+
+          if (response !== null && response) {
+            resolve(response.getReporterList())
+          }
+        },
+      )
     })
+  }
 
-    let code = 0
-    let errorLog
-
-    if (response.code) {
-      code = parseInt(response.code)
-      errorLog = response.raw_log
-    }
+  async getLatestBlock(): Promise<GetLatestBlockResponse.AsObject> {
+    const request = new GetLatestBlockRequest()
+    return new Promise((resolve, reject) => {
+      this.serviceClient.getLatestBlock(
+        request,
+        {} as grpc.Metadata,
+        (err, response) => {
+          if (err !== null) {
+            reject(err)
+            return
+          }
+
+          if (response !== null && response) {
+            resolve(response.toObject())
+          }
+        },
+      )
+    })
+  }
 
-    return {
-      txHash: Buffer.from(response.txhash, 'hex'),
-      code,
-      errorLog,
-    }
+  async getAccount(address: string): Promise<BaseAccount.AsObject> {
+    const request = new QueryAccountRequest()
+    request.setAddress(address)
+    return new Promise((resolve, reject) => {
+      this.authQueryClient.account(
+        request,
+        {} as grpc.Metadata,
+        (err, response) => {
+          if (err !== null || response === null) {
+            reject(err)
+            return
+          }
+          if (response.hasAccount()) {
+            let accBaseAccount = response
+              .getAccount()
+              .unpack(
+                BaseAccount.deserializeBinary,
+                'cosmos.auth.v1beta1.BaseAccount',
+              )
+            if (accBaseAccount.getAddress) {
+              resolve(accBaseAccount.toObject())
+            }
+          }
+          resolve(undefined)
+        },
+      )
+    })
   }
 
-  /**
-   * sendTxAsyncMode
-   * @param data
-   */
-  async sendTxAsyncMode(data: object): Promise<TransactionAsyncMode> {
-    let respose = await this.post('/txs', {
-      tx: data,
-      mode: 'async',
+  async getRequestIdByTxHash(txHash: string): Promise<any> {
+    const request = new GetTxRequest()
+    request.setHash(txHash)
+    return new Promise((resolve, reject) => {
+      this.txServiceClient.getTx(
+        request,
+        {} as grpc.Metadata,
+        (err, response) => {
+          if (err !== null) {
+            reject(err)
+            return
+          }
+          if (
+            response !== null &&
+            response.hasTxResponse() &&
+            response.toObject().txResponse.logsList
+          ) {
+            let reqIdList = []
+            for (
+              let z = 0;
+              z < response.toObject().txResponse.logsList.length;
+              z++
+            ) {
+              let tx = response.toObject().txResponse.logsList[z]
+              for (let x = 0; x < tx.eventsList.length; x++) {
+                if (
+                  tx.eventsList[x].type == 'report' ||
```

Comment:
I think for `report` type, attribute of request ID is named `requestId` but not `id` isn't it?

Reply:
For me it is `id`. Can you recheck with this txHash: `FF88A361014A8CC2283961D632F2A00BB9CC8C168D8AE7F307F50E5D8A3945D2` or maybe I did something wrong.

Reply:
Ah yes, it is `id`.